doc/classes/Array.xml

```diff
@@ -297,88 +303,83 @@
+				GD.Print(arr.Contains("inside"));  // Prints true
+				GD.Print(arr.Contains("outside")); // Prints false
+				GD.Print(arr.Contains(7));         // Prints true
+				GD.Print(arr.Contains("7"));       // Prints false
 				[/csharp]
 				[/codeblocks]
+				In GDScript, this is equivalent to the [code]in[/code] operator:
+				[codeblock]
+				if 4 in [2, 4, 6, 8]:
+				    print("There it is!")
+				[/codeblock]
+				[b]Note:[/b] For performance reasons, the search is affected by the [param value]'s [enum Variant.Type]. For example, [code]7[/code] ([int]) and [code]7.0[/code] ([float]) are not considered equal for this method.
 			</description>
 		</method>
 		<method name="hash" qualifiers="const">
 			<return type="int" />
 			<description>
 				Returns a hashed 32-bit integer value representing the array and its contents.
-				[b]Note:[/b] [Array]s with equal content will always produce identical hash values. However, the reverse is not true. Returning identical hash values does [i]not[/i] imply the arrays are equal, because different arrays can have identical hash values due to hash collisions.
+				[b]Note:[/b] Arrays with equal hash values are [i]not[/i] guaranteed to be the same, as a result of hash collisions. On the countrary, arrays with different hash values are guaranteed to be different.
 			</description>
 		</method>
 		<method name="insert">
 			<return type="int" />
 			<param index="0" name="position" type="int" />
 			<param index="1" name="value" type="Variant" />
 			<description>
-				Inserts a new element at a given position in the array. The position must be valid, or at the end of the array ([code]pos == size()[/code]).
-				[b]Note:[/b] This method acts in-place and doesn't return a value.
-				[b]Note:[/b] On large arrays, this method will be slower if the inserted element is close to the beginning of the array (index 0). This is because all elements placed after the newly inserted element have to be reindexed.
+				Inserts a new element ([param value]) at a given index ([param position]) in the array. [param position] should be between [code]0[/code] and the array's [method size].
+				?? [b]Note:[/b] The index of every element after [param position] is shifted forward. This means that this method is slower the closer [param position] is to, which may be especially noticeable on very large arrays.
+				?? [b]Note:[/b] This method is slower the closer [param position] is to the beginning of the array. This is because every element's index after [param position] is shifted forward, which may be especially slow on very large arrays.
+				?? [b]Note:[/b] This method shifts every element's index after [param position] forward, which may have a noticeable performance cost, especially on larger arrays.
+				?? [b]Note:[/b] Every element's index after [param position] needs to be shifted forward, which may have a noticeable performance cost, especially on larger arrays.
```

Comment:
Yes. 5 whole variations for this sentence. The last is the one I used elsewhere too because it is the simplest to understand in concept, but I'm not sure if that would be losing too much information.